applications/feedback_sentiment_analyzer/SPECIFICATION.md

```diff
@@ -0,0 +1,383 @@
+
+<details>
+<summary><strong>PUT /api/media/{item}</strong></summary>
+<br/>
+
+> Create or update an object in Amazon S3. Creating or updating an image file will trigger the Step Function.
+
+<table>
+<tr>
+<th>parameters</th>
+<th>request body</th>
+<th>response body</th>
+</tr>
+<tr>
+<td>item - the object key of the item to get</td>
+<td>JavaScript [File](https://developer.mozilla.org/en-US/docs/Web/API/File) object</td>
+<td>Empty</td>
+</tr>
+</table>
+</details>
+
+<details>
+<summary><strong>GET /api/media/{item}</strong></summary>
+<br/>
+
+> Get an object from Amazon S3.
+
+<table>
+<tr>
+<th>parameters</th>
+<th>request body</th>
+<th>response body</th>
+</tr>
+<tr>
+<td>item - the object key of the item to get</td>
+<td>Empty</td>
+<td>Empty</td>
+</tr>
+</table>
+</details>
+
+<details>
+<summary><strong>GET /api/feedback</strong></summary>
+<br/>
+
+> Get the translated text, sentiment, and audio/image keys for an uploaded image. This data comes from Amazon DynamoDB. The database table is filled as a result of running the step function.
+
+<table>
+<tr>
+<th>parameters</th>
+<th>request body</th>
+<th>response body</th>
+</tr>
+<tr>
+<td>Empty</td>
+<td>Empty</td>
+<td>
+<pre>
+{
+  "feedback": [
+    {
+      "sentiment": "POSITIVE",
+      "text": "I love this hotel!",
+      "audioUrl": "PXL_20230710_182358532.jpg.mp3",
+      "imageUrl": "PXL_20230710_182358532.jpg"
+    }
+  ]
+}
+</pre>
+</td>  
+</tr>
+</table>
+</details>
+
+<details>
+<summary><strong>GET /api/env</strong></summary>
+<br/>
+
+> Get the environment variables required to connect to a Cognito hosted UI. The frontend calls this automatically to facilitate sign in.
+
+<table>
+<tr>
+<th>parameters</th>
+<th>request body</th>
+<th>response body</th>
+</tr>
+<tr>
+<td>Empty</td>
+<td>Empty</td>
+<td>
+<pre>
+{
+"COGNITO_SIGN_IN_URL": "https://...",
+"COGNITO_SIGN_OUT_URL": "https://..."
+}
+</pre>
+</td>
+</tr>
+</table>
+</details>
+
+## Step Function configuration
+
+When an image is created or updated in S3 media bucket, a Step Function state machine is triggered.
+
+This multi-state workflow is listed below in sequence:
+
+1. Start
+2. [ExtractText](#ExtractText) - extracts text from an image
+3. [AnalyzeSentiment](#AnalyzeSentiment) - detects text sentiment
+4. `ContinueIfPositive` (skip to 7 if sentiment `NEGATIVE`)
+5. [TranslateText](#TranslateText) - translates text to English
+6. [SynthesizeAudio](#SynthesizeAudio) - synthesizes human-like audio from text
+7. `DynamoDB:PutItem` (see [table config](#managing-items-in-dynamodb))
+8. Stop
+
+The below diagram depicts this sequence.
+![state-machine.png](state-machine.png)
+
+## State machine Lambda functions
+
+See below for the required inputs and outputs of each Lambda function.
+
+### ExtractText
+Uses Amazon Textract's [DetectDocumentText](https://docs.aws.amazon.com/textract/latest/dg/API_DetectDocumentText.html)
```

Comment:
```suggestion
Uses the Amazon Textract [DetectDocumentText](https://docs.aws.amazon.com/textract/latest/dg/API_DetectDocumentText.html)
```

Reply:
Sigh. Apologies. I just looked at it wrong.